### PR TITLE
Feature : Add in app update support.

### LINKED ISSUE #475: Add In App Update support.
**Expected Behaviour**  
On App start, there should be a prompt to update the application if newer version is available.

**Steps to reproduce it**  
NA

**Additional context**  
NA

**Would you like to work on the issue?**
Yes

### PR DESCRIPTION
**Pull request by dipu989**
Fixes #475 

**Changes**: 
Added In App update support.
If new version of app is available, user is prompted to download the updates.
Reference : - https://developer.android.com/guide/app-bundle/in-app-updates

**Screenshot/s for the changes**:
NA

**Checklist**:
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[apkDebug.zip](https://github.com/fossasia/neurolab-android/files/3640641/apkDebug.zip)

### CODE REVIEW

**Comment by dipu989** [comment]
I have formatted the code and checked for any Unused imports as well, even though Codacy checks are failing!

File: app/src/main/java/io/neurolab/main/NeuroLab.java [hunk 1]
```diff
@@ -175,8 +188,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                                            this,
+                                            MY_REQUEST_CODE);
+                                } catch (IntentSender.SendIntentException e) {
+                                    e.printStackTrace();
+                                }
+                            }
+                        });
+    }
+
+
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/NeuroLab.java]: Extra line. Please remove.

File: app/src/main/java/io/neurolab/main/NeuroLab.java [hunk 2]
```diff
@@ -175,8 +188,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
         }
+
+        checkForUpdates(appUpdateManager, appUpdateInfoTask);
     }
 
+    private void checkForUpdates(AppUpdateManager appUpdateManager, Task<AppUpdateInfo> appUpdateInfoTask) {
+        appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE && appUpdateInfo.isUpdateTypeAllowed(IMMEDIATE)) {
+                try {
+
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/NeuroLab.java]: Extra line. Remove.

File: app/src/main/java/io/neurolab/main/NeuroLab.java [hunk 3]
```diff
@@ -175,8 +188,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                    appUpdateManager.startUpdateFlowForResult(appUpdateInfo, IMMEDIATE, this, MY_REQUEST_CODE);
+
+                } catch (IntentSender.SendIntentException e) {
+                    e.printStackTrace();
+                }
+
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/NeuroLab.java]: Remove

File: app/src/main/java/io/neurolab/main/NeuroLab.java [hunk 4]
```diff
@@ -175,8 +188,58 @@ protected void onCreate(Bundle savedInstanceState) {
...
+        checkForUpdates(appUpdateManager, appUpdateInfoTask);
     }
 
+    private void checkForUpdates(AppUpdateManager appUpdateManager, Task<AppUpdateInfo> appUpdateInfoTask) {
+        appUpdateInfoTask.addOnSuccessListener(appUpdateInfo -> {
+            if (appUpdateInfo.updateAvailability() == UpdateAvailability.UPDATE_AVAILABLE && appUpdateInfo.isUpdateTypeAllowed(IMMEDIATE)) {
+                try {
+
+                    appUpdateManager.startUpdateFlowForResult(appUpdateInfo, IMMEDIATE, this, MY_REQUEST_CODE);
+
```
Comment (jddeep) [inline app/src/main/java/io/neurolab/main/NeuroLab.java]: Remove

**Review by jddeep — CHANGES_REQUESTED** [review]

**Comment by dipu989** [comment]
@jddeep Removed the extra lines.

**Review by jddeep — APPROVED** [review]

**Comment by dipu989** [comment]
@adityastic  Can you please review this PR?

**Review by adityastic — APPROVED** [review]
Good job

File: app/src/main/java/io/neurolab/main/NeuroLab.java [hunk 5]
```diff
@@ -175,6 +188,52 @@ protected void onCreate(Bundle savedInstanceState) {
...
+                } catch (IntentSender.SendIntentException e) {
+                    e.printStackTrace();
+                }
+            }
+        });
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if (requestCode == MY_REQUEST_CODE) {
```
Comment (adityastic) [inline app/src/main/java/io/neurolab/main/NeuroLab.java]: Combine both the conditions. There is no else, hence no need to create separate conditions

Comment (dipu989) [reply app/src/main/java/io/neurolab/main/NeuroLab.java]: @adityastic Sure! On it!

Comment (dipu989) [reply app/src/main/java/io/neurolab/main/NeuroLab.java]: @adityastic  Done!

**Pull request merged**